pointnetwork/pointnetwork-dashboard
# Referral Code Program

This PR gets the referral code from the downloaded binary and saves it `~/.point/infoReferral.json`.

Then once the user registers their identity, we ping the bounty server to claim the reward.

**HOW TO TEST**
1. Stop any running instances of dashboard and node
2. `rm -rf ~/.point`
3. Download the binary from the GitHub release page. If you already have downloaded one previously, then skip this step.
4. Append `-123456789012` or any 12 digit number to the binary name. For example: `point-v0.2.17-Linux-Debian-Ubuntu-123456789012.tar.gz`
5. Run `npm start` and go through the installation.
6. Once the installation is done, there should be a `infoReferral.json` file in the `~/.point` directory. Within that file, there should be the `referralCode:123456789012` or whatever 12-digit code you used. Also, if the request to bounty server with `event=installed` is successful, then there will also be `isInstalledEventSent:true`.
7. Next, generate a seed phrase and login but don't register an identity yet. Check the `infoReferral.json` file again. If the request to bounty server with `event=generated` is successful, then there will also be `isGeneratedEventSent:true`.

The `referralCode` will and should be null if the code is less than 12-digits or if there's no 12-digit code appended.
